internal/juju/common_test.go

```diff
@@ -13,14 +13,14 @@ import (
 type JujuSuite struct {
 	suite.Suite
 
-	testModelName string
+	testModelName *string
 
 	mockConnection   *MockConnection
 	mockSharedClient *MockSharedClient
 }
 
-func (s *JujuSuite) setupMocks(t *testing.T) *gomock.Controller {
-	s.testModelName = "test-secret-model"
+func (s *JujuSuite) setupMocks(t *testing.T, modelName *string) *gomock.Controller {
+	s.testModelName = modelName
```

Comment:
issue: This 2 line change is not needed. Instead the secret_test.go file should be changed
```diff
diff --git a/internal/juju/secrets_test.go b/internal/juju/secrets_test.go
index e74be5b..0688cfc 100644
--- a/internal/juju/secrets_test.go
+++ b/internal/juju/secrets_test.go
@@ -19,17 +19,15 @@ type SecretSuite struct {
        suite.Suite
        JujuSuite

-       testModelName string
-
        mockSecretClient *MockSecretAPIClient
 }

-func (s *SecretSuite) SetupTest() {}
+func (s *SecretSuite) SetupSuite() {
+       s.testModelName = strPtr("test-secret-model")
+}

 func (s *SecretSuite) setupMocks(t *testing.T) *gomock.Controller {
-       s.testModelName = "test-secret-model"
-
-       ctlr := s.JujuSuite.setupMocks(t, &s.testModelName)
+       ctlr := s.JujuSuite.setupMocks(t)
        s.mockSecretClient = NewMockSecretAPIClient(ctlr)

        return ctlr
@@ -60,7 +58,7 @@ func (s *SecretSuite) TestCreateSecret() {

        client := s.getSecretsClient()
        output, err := client.CreateSecret(&CreateSecretInput{
-               ModelName: s.testModelName,
+               ModelName: *s.testModelName,
                Name:      "test-secret",
                Value:     decodedValue,
                Info:      "test info",
```